File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/EditorScroller.java

package io.github.abc15018045126.sora.widget;

import android.widget.OverScroller;

import androidx.annotation.NonNull;

public class EditorScroller {

    private final CodeEditor editor;

    private final OverScroller scroller;

    public EditorScroller(@NonNull CodeEditor editor) {
        scroller = new OverScroller(editor.getContext());
        this.editor = editor;
    }

    public void setEditorOffsets() {
        editor.setScrollX(scroller.getCurrX());
        editor.setScrollY(scroller.getCurrY());
    }

    public void startScroll(int startX, int startY, int dx, int dy) {
        startScroll(startX, startY, dx, dy, editor.getProps().scrollAnimationDurationMs);
    }

    public void startScroll(int startX, int startY, int dx, int dy, int duration) {
        scroller.startScroll(startX, startY, dx, dy, duration);
        setEditorOffsets();
    }

    public void forceFinished(boolean finished) {
        scroller.forceFinished(finished);
        setEditorOffsets();
    }

    public void abortAnimation() {
        scroller.abortAnimation();
        setEditorOffsets();
    }

    public boolean isFinished() {
        return scroller.isFinished();
    }

    public int getCurrX() {
        return scroller.getCurrX();
    }

    public int getCurrY() {
        return scroller.getCurrY();
    }

    public int getFinalX() {
        return scroller.getFinalX();
    }

    public int getFinalY() {
        return scroller.getFinalY();
    }

    public int getStartX() {
        return scroller.getStartX();
    }

    public int getStartY() {
        return scroller.getStartY();
    }

    public float getCurrVelocity() {
        return scroller.getCurrVelocity();
    }

    public boolean computeScrollOffset() {
        var computed = scroller.computeScrollOffset();
        if (computed) {
            setEditorOffsets();
        }
        return computed;
    }

    public void fling(int startX, int startY, int velocityX, int velocityY,
                      int minX, int maxX, int minY, int maxY, int overX, int overY) {
        scroller.fling(startX, startY, velocityX, velocityY, minX, maxX, minY, maxY, overX, overY);
        setEditorOffsets();
    }

    public boolean isOverScrolled() {
        return scroller.isOverScrolled();
    }

    public OverScroller getImplScroller() {
        return scroller;
    }
}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/EditorScroller.kt
package io.github.abc15018045126.sora.widget

import android.widget.OverScroller
import androidx.annotation.NonNull

class EditorScroller(@NonNull private val editor: CodeEditor) {

    private val scroller: OverScroller = OverScroller(editor.context)

    fun setEditorOffsets() {
        editor.scrollX = scroller.currX
        editor.scrollY = scroller.currY
    }

    fun startScroll(startX: Int, startY: Int, dx: Int, dy: Int) {
        startScroll(startX, startY, dx, dy, editor.props!!.scrollAnimationDurationMs)

    }

    fun startScroll(startX: Int, startY: Int, dx: Int, dy: Int, duration: Int) {
        scroller.startScroll(startX, startY, dx, dy, duration)
        setEditorOffsets()
    }

    fun forceFinished(finished: Boolean) {
        scroller.forceFinished(finished)
        setEditorOffsets()
    }

    fun abortAnimation() {
        scroller.abortAnimation()
        setEditorOffsets()
    }

    val isFinished: Boolean
        get() = scroller.isFinished

    fun getCurrX(): Int {
        return scroller.currX
    }

    fun getCurrY(): Int {
        return scroller.currY
    }

    fun getFinalX(): Int {
        return scroller.finalX
    }

    fun getFinalY(): Int {
        return scroller.finalY
    }

    fun getStartX(): Int {
        return scroller.startX
    }

    fun getStartY(): Int {
        return scroller.startY
    }

    fun getCurrVelocity(): Float {
        return scroller.currVelocity
    }

    fun computeScrollOffset(): Boolean {
        val computed = scroller.computeScrollOffset()
        if (computed) {
            setEditorOffsets()
        }
        return computed
    }

    fun fling(
        startX: Int, startY: Int, velocityX: Int, velocityY: Int,
        minX: Int, maxX: Int, minY: Int, maxY: Int, overX: Int, overY: Int
    ) {
        scroller.fling(startX, startY, velocityX, velocityY, minX, maxX, minY, maxY, overX, overY)
        setEditorOffsets()
    }

    fun isOverScrolled(): Boolean {
        return scroller.isOverScrolled
    }

    fun getImplScroller(): OverScroller {
        return scroller
    }
}
